crates/pruning/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+use criteria::DistanceCriteria;
+use futures::future;
+use serde::{Deserialize, Serialize};
+use sov_db::ledger_db::SharedLedgerOps;
+use tokio::select;
+use tokio::sync::broadcast;
+use tokio_util::sync::CancellationToken;
+use tracing::{error, info};
+
+use crate::criteria::Criteria;
+use crate::pruners::{prune_evm, prune_ledger};
+
+mod criteria;
+mod pruners;
+#[cfg(test)]
+mod tests;
+
+/// A configuration type to define the behaviour of the pruner.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct PruningConfig {
+    /// Defines the number of blocks from the tip of the chain to remove.
+    pub distance: u64,
+}
+
+impl Default for PruningConfig {
+    fn default() -> Self {
+        Self { distance: 256 }
+    }
+}
+
+pub struct Pruner<DB>
```

Comment:
direct access to NativeStorage, or StorageManager might be helpful here.

Reply:
can be added later ofc

Reply:
Yea i'd like the later ticket implementer to add this